pkg/autoscaler/metrics/stats_scraper.go

```diff
@@ -150,23 +151,23 @@ func newServiceScraperWithClient(
 		return nil, err
 	}
 
-	return &ServiceScraper{
+	return &serviceScraper{
 		sClient:  sClient,
 		counter:  counter,
 		url:      urlFromTarget(metric.Spec.ScrapeTarget, metric.ObjectMeta.Namespace),
 		statsCtx: ctx,
 	}, nil
 }
 
+var portAndPath = strconv.Itoa(networking.AutoscalingQueueMetricsPort) + "/metrics"
+
 func urlFromTarget(t, ns string) string {
-	return fmt.Sprintf(
-		"http://%s.%s:%d/metrics",
-		t, ns, networking.AutoscalingQueueMetricsPort)
+	return fmt.Sprintf("http://%s.%s:", t, ns) + portAndPath
```

Comment:
since we're micro-optimising a bit anyway (:D), looks like it avoids a little bit of allocation and is slightly faster to do the whole thing in the sprintf rather than a sprintf and a concatenate (pre-computing the port and path is still a win either way):

```suggestion
	return fmt.Sprintf("http://%s.%s:%s", t, ns, portAndPath)
```

Reply:
(may potentially be even faster to use a builder or a straight concatenate of the whole thing I guess since everything is strings now, but now we're really off the deep end so 🤷‍♂️)

Reply:
```
benchmark            old ns/op     new ns/op     delta
BenchmarkTest-12     217           193           -11.06%
benchmark            old allocs     new allocs     delta
BenchmarkTest-12     3              3              +0.00%
benchmark            old bytes     new bytes     delta
BenchmarkTest-12     64            64            +0.00%
```

Allocation behavior doesn't change between the two, though the concatenation seems slightly faster. I'd opt for readability here as this is called quite rarely anyway. Not on a hot path.

Reply:
hm, I get 

```
goarch: amd64
BenchmarkSprintfConcat-16        5142325               230 ns/op             112 B/op          4 allocs/op
BenchmarkSprintf-16              5567184               216 ns/op              96 B/op          4 allocs/op
BenchmarkConcat-16              15808464                77.4 ns/op            48 B/op          1 allocs/op
PASS
```

so very slightly (and I totally don't care either honestly :)) less bytes allocated per op. 100% agree to prefer readability though, I'd personally argue a single sprintf is a little easier to read the sprintf + concat, but to say I don't feel strongly about this would be a significant understatement :).

Reply:
Yeah, I opted to not comment 😂. It's fine to leave as is and I guess the behavior is changing with per-pod scraping so it's fine to optimize the static path now.

Reply:
haha I regretted it about 5 seconds after I did!

Reply:
We need to see the benchmarks as well before the judges can return with best benchmark prize though. 😀